tests/d3d12_pso.c

```diff
@@ -3959,3 +3959,490 @@ void test_view_instancing(void)
+            &vs_mismatch_dxil, &ps_mismatch_ref_dxil, NULL);
+    pso_desc.NumRenderTargets = render_target_subobject.render_target_formats.NumRenderTargets;
+
+    for (i = 0; i < pso_desc.NumRenderTargets; i++)
+        pso_desc.RTVFormats[i] = render_target_subobject.render_target_formats.RTFormats[i];
+
+    for (i = 0; i < ARRAY_SIZE(input_descs); i++)
+    {
+        vkd3d_test_set_context("Test %u", i);
+
+        pso_desc.InputLayout.NumElements = 1;
+        pso_desc.InputLayout.pInputElementDescs = &input_descs[i];
+
+        /* Types aren't validated, but fail if a semantic is not provided */
+        expected = i > 1 ? E_INVALIDARG : S_OK;
+
+        hr = ID3D12Device_CreateGraphicsPipelineState(context.device, &pso_desc, &IID_ID3D12PipelineState, (void **)&pso);
+        ok(hr == expected, "Got hr %#x, expected %#x.\n", hr, expected);
+
+        if (SUCCEEDED(hr))
+            ID3D12PipelineState_Release(pso);
+    }
+
+    /* Test general shader interface compatibility */
+    pso_desc.InputLayout.NumElements = 1;
+    pso_desc.InputLayout.pInputElementDescs = &input_descs[0];
+
+    for (i = 0; i < ARRAY_SIZE(tests); i++)
+    {
+        vkd3d_test_set_context("Test %u", i);
+
+        if (!tests[i].vs)
+            continue;
+
+        if ((supported_features & tests[i].required_features) != tests[i].required_features)
+        {
+            skip("Features %#x not supported.\n", tests[i].required_features);
+            continue;
+        }
+
+        expected = tests[i].vs_should_compile ? S_OK : E_INVALIDARG;
+
+        memset(&pso_desc.HS, 0, sizeof(pso_desc.HS));
+        memset(&pso_desc.DS, 0, sizeof(pso_desc.DS));
+        memset(&pso_desc.GS, 0, sizeof(pso_desc.GS));
+        memset(&pso_desc.PS, 0, sizeof(pso_desc.PS));
+
+        pso_desc.VS = *tests[i].vs;
+
+        if (tests[i].hs)
+        {
+            pso_desc.HS = *tests[i].hs;
+            pso_desc.DS = *tests[i].ds;
+
+            pso_desc.PrimitiveTopologyType = D3D12_PRIMITIVE_TOPOLOGY_TYPE_PATCH;
+        }
+        else
+            pso_desc.PrimitiveTopologyType = D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE;
+
+        if (tests[i].gs)
+            pso_desc.GS = *tests[i].gs;
+
+        if (tests[i].ps)
+            pso_desc.PS = *tests[i].ps;
+
+        hr = ID3D12Device_CreateGraphicsPipelineState(context.device, &pso_desc, &IID_ID3D12PipelineState, (void **)&pso);
+        ok(hr == expected, "Got hr %#x, expected %#x.\n", hr, expected);
+
+        if (SUCCEEDED(hr))
+            ID3D12PipelineState_Release(pso);
+    }
+
+    /* Test some stream output scenarios with or without rasterized stream */
+    pso_desc.PrimitiveTopologyType = D3D12_PRIMITIVE_TOPOLOGY_TYPE_TRIANGLE;
+
+    memset(&pso_desc.HS, 0, sizeof(pso_desc.HS));
+    memset(&pso_desc.DS, 0, sizeof(pso_desc.DS));
+
+    for (i = 0; i < ARRAY_SIZE(so_tests); i++)
+    {
+        vkd3d_test_set_context("Test %u", i);
+
+        expected = so_tests[i].should_compile ? S_OK : E_INVALIDARG;
```

Comment:
Where do we validate stream-out signature compat?

Reply:
We kind of don't really, I could add that too (as well as more tests), but the intention of this particular test was to see how SO and especially the rasterized stream index interacts with interface compatibility (FS is removed if rasterization is disabled, which we already did, and otherwise it's kind of broken).